src/Resolver/ResolverMapRegistry.php

```diff
@@ -0,0 +1,236 @@
+     */
+    protected function map(): array
+    {
+        if ($this->resolverMap === null) {
+            $this->resolverMap = $this->loadResolverMap();
+        }
+        $map = [];
+        foreach ($this->resolverMap as $typeName => $fieldMap) {
+            $map[$typeName] = [
+                self::RESOLVE_FIELD => $this->buildResolverFunction($fieldMap)
+            ];
+        }
+
+        $map['Teaser'] = [
+            self::RESOLVE_TYPE => function ($value) {
+                return $this->resolveType($value);
+            }
+        ];
+        $map['Asset'] = [
+            self::RESOLVE_TYPE => function ($value) {
+                return $this->resolveType($value);
+            }
+        ];
+        return $map;
+    }
+
+    private function resolveType(object $value): string
+    {
+        $className = get_class($value);
+        return (substr($className, strrpos($className, '\\') + 1));
+    }
+
+    /**
+     * @return array<string,array<string,ResolverMethod>>
+     */
+    private function loadResolverMap(): array
+    {
+        $mapList = [];
+        foreach ($this->resolverList as $resolver) {
+            $mapList[] = $this->getResolverGetterMap($resolver);
+        }
+        return array_merge_recursive(
+            ...$mapList
+        );
+    }
+
+    /**
+     * @param array<string,ResolverMethod> $fieldMap
+     */
+    private function buildResolverFunction(array $fieldMap): Closure
+    {
+        return function (
+            $value,
+            ArgumentInterface $args,
+            ArrayObject $context,
+            ResolveInfo $info
+        ) use ($fieldMap) {
+            $fieldName = $info->fieldName;
+            if (isset($fieldMap[$fieldName])) {
+                $resolverMethod = $fieldMap[$fieldName];
+                return $this->invokeGetter(
+                    $resolverMethod,
+                    $value,
+                    $args
+                );
+            }
+
+            return $this->resolveProperty($value, $fieldName);
+        };
+    }
+
+    /**
+     * @throws ReflectionException
+     */
+    private function invokeGetter(
+        ResolverMethod $resolverMethod,
+        object $obj,
+        ArgumentInterface $args
+    ): mixed {
+        $method = $resolverMethod->reflectionMethod;
+        $params = $method->getParameters();
+        $paramLength = count($params);
+        if ($paramLength === 1) {
+            return $method->invoke($resolverMethod->resolver, $obj);
+        }
+
+        return $method->invoke($resolverMethod->resolver, $obj, $args);
+    }
+
+    /**
+     * @throws ReflectionException
+     */
+    private function resolveProperty(
+        object $obj,
+        string $fieldName
+    ): mixed {
+        $reflectionClass = new ReflectionClass(get_class($obj));
+        return $reflectionClass
+            ->getProperty($fieldName)
+            ->getValue($obj);
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
```

Comment:
PHPStorm sagt:

> Exception 'ReflectionException' is never thrown in the function

This is probably because `new ReflectionClass()` only throws it if the class does not exist. But this can never be the case:

```php
        $class = new ReflectionClass(get_class($resolver));
```